org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java

```diff
@@ -691,74 +704,82 @@ private void collectAttributeValueSuggestions(int valueStart, int valueEnd, Comp
 	}
 
 	private void collectInsideDTDContent(CompletionRequest request, CompletionResponse response) {
+		collectInsideDTDContent(request,response, false);
+	}
+	private void collectInsideDTDContent(CompletionRequest request, CompletionResponse response, boolean isContent) {
 		// Insert DTD Element Declaration
 		// see https://www.w3.org/TR/REC-xml/#dt-eldecl
 		CompletionItem elementDecl = new CompletionItem();
 		elementDecl.setLabel("Insert DTD Element declaration");
 		elementDecl.setKind(CompletionItemKind.EnumMember);
-		elementDecl.setFilterText("<!ELEMENT");
+		elementDecl.setFilterText("<!ELEMENT ");
 		elementDecl.setInsertTextFormat(InsertTextFormat.Snippet);
 		int startOffset = request.getOffset();
+		Range editRange = null;
+		DOMDocument document = request.getXMLDocument();
+		DOMNode node = document.findNodeAt(startOffset);
 		try {
-			Range editRange = getReplaceRange(startOffset, startOffset, request);
-			elementDecl.setTextEdit(new TextEdit(editRange, "<!ELEMENT ${0:element-name} (${1:#PCDATA})>"));
-			elementDecl.setDocumentation("<!ELEMENT element-name (#PCDATA)>");
+			if(node.isDoctype()) {
+				editRange = getReplaceRange(startOffset, startOffset, request);
+			} else {
+				if(isContent) {
+					editRange = document.getTrimmedRange(node.getStart(), node.getEnd());
+				} 
+				if(editRange == null) {
+					editRange = getReplaceRange(node.getStart(), node.getEnd(), request);
+				}
+			}
 		} catch (BadLocationException e) {
-			LOGGER.log(Level.SEVERE, "While performing getReplaceRange for DTD !ELEMENT completion.", e);
+			LOGGER.log(Level.SEVERE, "While performing getReplaceRange for DTD completion.", e);
 		}
+				
+		elementDecl.setTextEdit(new TextEdit(editRange, "<!ELEMENT ${0:element-name} (${1:#PCDATA})>"));

```

Comment:
need to check if snippets supported by client. 

Also, for some reason vscode selects placeholder 1 instead of 0. Check if using 1 and 2 instead works better